organize_flowcell.py

```diff
@@ -0,0 +1,206 @@
+                    row[sample_name_i],
+                )
+                include = not (
+                    lane in exclude_lane
+                    or sample_id in exclude_sampleID
+                    or sample_name in exclude_sample
+                )
+
+                if include:
+                    description = dict(
+                        item.split(":") for item in row[description_i].split(";")
+                    )
+                    library_name = description["LIBRARY_NAME"]
+                    sample_info.setdefault(sample_id, {})[sample_name] = [
+                        lane,
+                        library_name,
+                    ]
+
+    except csv.Error as e:
+        print(f"Error parsing SampleSheet.csv: {e}")
+
+    return sample_info
+
+
+def organize_files(sample_info, runfolder_path, project, data_path):
+    runfolder = os.path.basename(runfolder_path)
+    for sample_id, samples in sample_info.items():
+        for sample_name, (lane, library_name) in samples.items():
+            fq_folder = os.path.join(runfolder_path, "Unaligned", project, sample_id)
+            src_path = glob(os.path.join(fq_folder, "*fastq.gz"))
+            if src_path:
+                dst_path = os.path.join(data_path, sample_name, library_name, runfolder)
+                os.makedirs(dst_path, exist_ok=True)
+                for fastq in src_path:
+                    symlink(dst_path, fastq)
+            else:
+                print(f"No fastq.gz files found in {fq_folder}")
+
+
+def symlink(dst_path, fastq):
+    if os.path.exists(fastq):
+        fq_name = os.path.basename(fastq)
+        dst_file = os.path.join(dst_path, fq_name)
+        try:
+            os.symlink(fastq, dst_file)
+        # Should catch possible permission issues
+        except OSError as e:
+            print(f"Error creating symlink: {e}")
+    else:
+        print(f"No symlink created for {fastq}. File not found.")
+
+
+def check_paths(runfolder_path, fastq_path, samplesheet, data_path, project, force):
+    if not os.path.exists(runfolder_path):
+        raise FileNotFoundError(f"{runfolder_path} not found.")
+    if not os.path.exists(fastq_path):
+        raise FileNotFoundError(f"Project {project} not found in {runfolder_path}.")
+    if not os.path.exists(samplesheet):
+        raise FileNotFoundError(f"No sample sheet found for {runfolder_path}.")
+
+    organized_data = glob(
+        os.path.join(data_path, "*/*", os.path.basename(runfolder_path))
+    )
+    if organized_data:
+        if force:
+            remove_organized(organized_data)
+        else:
+            raise Exception("Flowcell already organized for this project.")
+
+
+def remove_organized(organized_data):
+    for path in organized_data:
+        # Remove folder and symlinks
+        shutil.rmtree(path)
+        clean_empty_parent_dirs(path)
+
+
+def clean_empty_parent_dirs(path):
+    parent_dir = os.path.dirname(path)
+    if not os.listdir(parent_dir):
```

Comment:
Maybe I'm overcautious, but it might be safer to do: `if len(os.listdir(parent_dir)) == 0:`

If the path for some reason is None, os.listdir lists the CWD instead.